src/main/java/org/fungover/mmotodo/rest/task/TaskController.java

```diff
@@ -0,0 +1,59 @@
+package org.fungover.mmotodo.rest.task;
+
+import org.fungover.mmotodo.entities.task.Task;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/tasks")
+public class TaskController {
+
+    private final TaskService service;
+
+    public TaskController(TaskService service) {
+        this.service = service;
+    }
+
+    @GetMapping("/")
+    public ResponseEntity<List<Task>> getAllTasks(@RequestParam(required = false) Long userId,
+                                                  @RequestParam(required = false) Long teamId) {
+        if (userId != null) {
+            return ResponseEntity.ok(service.findTasksByUserId(userId));
+        } else if (teamId != null) {
+            return ResponseEntity.ok(service.findTasksByTeamId(teamId));
+        }
+        return ResponseEntity.ok(service.findAllTasks());
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Task> getTaskById(@PathVariable Long id) {
+        Task task = service.findTaskById(id);
+        return task != null ? ResponseEntity.ok(task) : ResponseEntity.notFound().build();
```

Comment:
If using the same TaskService that's in main branch, the `getTaskById` already throws a `TaskNotFoundException` with a nice json response so there's really no need to handle "notFound" here! :)
```java
return service.getTaskById(id);
```